domain/src/main/java/org/oppia/android/domain/question/QuestionSessionMetrics.kt

```diff
@@ -0,0 +1,14 @@
+package org.oppia.android.domain.question
+import org.oppia.android.app.model.Question
+
+/**
+ * Mutable data class for tracking a user's interactions with a specific question. This includes
+ * the number of answers they submitted, the number of hints they used, whether they viewed the
+ * solution, and the skill misconception ids related to their wrong answers.
+ */
+data class QuestionSessionMetrics(val question: Question) {
```

Comment:
Can we make this ``internal`` to ensure it can't be used broadly? Similar to StateDeck & other utilities used only for internal processing.